Support linux arm64/aarch64 architecture

The releases for arm64/aarch64 are already provided on the Protobuf releases page https://github.com/protocolbuffers/protobuf/releases and thus can easily be supported

I tried working on this myself, but I can't run `npm run build`:

```sh
% npm run build

> setup-protoc-action@0.0.0 build
> tsc

node_modules/keyv/src/index.d.ts:60:32 - error TS2304: Cannot find name 'AsyncGenerator'.

60  iterator(namespace?: string): AsyncGenerator<any, void, any>;
                                  ~~~~~~~~~~~~~~


Found 1 error.
```

Any ideas? I'm on WSFL, Ubuntu, using node v16.17.1

--- a/__tests__/main.test.ts
+++ b/__tests__/main.test.ts
@@ -14,6 +14,24 @@ process.env["RUNNER_TEMP"] = tempDir;
 process.env["RUNNER_TOOL_CACHE"] = toolDir;
 import * as installer from "../src/installer";
 
+describe("filename tests", () => {
+  const tests = [
+    ["protoc-3.20.2-linux-x86_32.zip", "linux", ""],
+    ["protoc-3.20.2-linux-x86_64.zip", "linux", "x64"],
+    ["protoc-3.20.2-linux-aarch_64.zip", "linux", "arm64"],
+    ["protoc-3.20.2-linux-ppcle_64.zip", "linux", "ppc64"],
+    ["protoc-3.20.2-linux-s390_64.zip", "linux", "s390x"],
+    ["protoc-3.20.2-osx-aarch_64.zip", "darwin", "arm64"],
+    ["protoc-3.20.2-osx-x86_64.zip", "darwin", "x64"]
+  ];
+  for (const [expected, plat, arch] of tests) {
+    it(`downloads ${expected} correctly`, () => {
+      const actual = installer.getFileName("3.20.2", plat, arch);
+      expect(expected).toBe(actual);
+    });
+  }
+});
+
 describe("installer tests", () => {
   beforeEach(async function() {
     await io.rmRF(toolDir);

--- a/__tests__/main.test.ts
+++ b/__tests__/main.test.ts
@@ -22,10 +22,9 @@ describe("filename tests", () => {
     ["protoc-3.20.2-linux-ppcle_64.zip", "linux", "ppc64"],
     ["protoc-3.20.2-linux-s390_64.zip", "linux", "s390x"],
     ["protoc-3.20.2-osx-aarch_64.zip", "darwin", "arm64"],
-    ["protoc-3.20.2-osx-x86_64.zip", "darwin", "x64"]
     ["protoc-3.20.2-osx-x86_64.zip", "darwin", "x64"],
     ["protoc-3.20.2-win64.zip", "win32", "x64"],
-    ["protoc-3.20.2-win32.zip", "win32", "x32"],
+    ["protoc-3.20.2-win32.zip", "win32", "x32"]
   ];
   for (const [expected, plat, arch] of tests) {
     it(`downloads ${expected} correctly`, () => {

--- a/__tests__/main.test.ts
+++ b/__tests__/main.test.ts
@@ -23,6 +23,9 @@ describe("filename tests", () => {
     ["protoc-3.20.2-linux-s390_64.zip", "linux", "s390x"],
     ["protoc-3.20.2-osx-aarch_64.zip", "darwin", "arm64"],
     ["protoc-3.20.2-osx-x86_64.zip", "darwin", "x64"]
+    ["protoc-3.20.2-osx-x86_64.zip", "darwin", "x64"],
+    ["protoc-3.20.2-win64.zip", "win32", "x64"],
+    ["protoc-3.20.2-win32.zip", "win32", "x32"],
   ];
   for (const [expected, plat, arch] of tests) {
     it(`downloads ${expected} correctly`, () => {

--- a/lib/installer.js
+++ b/lib/installer.js
@@ -88,7 +88,7 @@ exports.getProtoc = getProtoc;
 function downloadRelease(version) {
     return __awaiter(this, void 0, void 0, function* () {
         // Download
-        let fileName = getFileName(version);
+        let fileName = getFileName(version, osPlat, osArch);
         let downloadUrl = util.format("https://github.com/protocolbuffers/protobuf/releases/download/%s/%s", version, fileName);
         process.stdout.write("Downloading archive: " + downloadUrl + os.EOL);
         let downloadPath = null;
@@ -105,7 +105,43 @@ function downloadRelease(version) {
         return yield tc.cacheDir(extPath, "protoc", version);
     });
 }
-function getFileName(version) {
+/**
+ *
+ * @param osArch - A string identifying operating system CPU architecture for which the Node.js binary was compiled.
+ * See https://nodejs.org/api/os.html#osarch for possible values.
+ * @returns Suffix for the protoc filename.
+ */
+function fileNameSuffix(osArch) {
+    switch (osArch) {
+        case "x64": {
+            return "x86_64";
+        }
+        case "arm64": {
+            return "aarch_64";
+        }
+        case "s390x": {
+            return "s390_64";
+        }
+        case "ppc64": {
+            return "ppcle_64";
+        }
+        default: {
+            return "x86_32";
+        }
+    }
+}
+/**
+ * Returns the filename of the protobuf compiler.
+ *
+ * @param version - The version to download
+ * @param osPlat - The operating system platform for which the Node.js binary was compiled.
+ * See https://nodejs.org/api/os.html#osplatform for more.
+ * @param osArch - The operating system CPU architecture for which the Node.js binary was compiled.
+ * See https://nodejs.org/api/os.html#osarch for more.
+ * @returns The filename of the protocol buffer for the given release, platform and architecture.
+ *
+ */
+function getFileName(version, osPlat, osArch) {
     // to compose the file name, strip the leading `v` char
     if (version.startsWith("v")) {
         version = version.slice(1, version.length);
@@ -115,12 +151,13 @@ function getFileName(version) {
         const arch = osArch == "x64" ? "64" : "32";
         return util.format("protoc-%s-win%s.zip", version, arch);
     }
-    const arch = osArch == "x64" ? "x86_64" : "x86_32";
+    const suffix = fileNameSuffix(osArch);
     if (osPlat == "darwin") {
-        return util.format("protoc-%s-osx-%s.zip", version, arch);
+        return util.format("protoc-%s-osx-%s.zip", version, suffix);
     }
-    return util.format("protoc-%s-linux-%s.zip", version, arch);
+    return util.format("protoc-%s-linux-%s.zip", version, suffix);
 }
+exports.getFileName = getFileName;
 // Retrieve a list of versions scraping tags from the Github API
 function fetchVersions(includePreReleases, repoToken) {
     return __awaiter(this, void 0, void 0, function* () {
@@ -135,8 +172,9 @@ function fetchVersions(includePreReleases, repoToken) {
         }
         let tags = [];
         for (let pageNum = 1, morePages = true; morePages; pageNum++) {
-            let nextPage = (yield rest.get("https://api.github.com/repos/protocolbuffers/protobuf/releases?page=" +
-                pageNum)).result || [];
+            let p = yield rest.get("https://api.github.com/repos/protocolbuffers/protobuf/releases?page=" +
+                pageNum);
+            let nextPage = p.result || [];
             if (nextPage.length > 0) {
                 tags = tags.concat(nextPage);
             }

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -91,7 +91,7 @@ export async function getProtoc(
 
 async function downloadRelease(version: string): Promise<string> {
   // Download
-  let fileName: string = getFileName(version);
+  let fileName: string = getFileName(version, osPlat, osArch);
   let downloadUrl: string = util.format(
     "https://github.com/protocolbuffers/protobuf/releases/download/%s/%s",
     version,
@@ -114,7 +114,48 @@ async function downloadRelease(version: string): Promise<string> {
   return await tc.cacheDir(extPath, "protoc", version);
 }
 
-function getFileName(version: string): string {
+/**
+ *
+ * @param osArch - A string identifying the operating system platform for which the Node.js binary was compiled.
+ * See https://nodejs.org/api/os.html#osplatform for possible values.
+ * @returns Suffix for the protoc filename.
+ */
+function fileNameSuffix(osArch: string): string {
+  switch (osArch) {
+    case "x64": {
+      return "x86_64";
+    }
+    case "arm64": {
+      return "aarch_64";
+    }
+    case "s390x": {
+      return "s390_64";
+    }
+    case "ppc64": {
+      return "ppcle_64";
+    }
+    default: {
+      return "x86_32";
+    }
+  }
+}
+
+/**
+ * Returns the filename of the protobuf compiler.
+ *
+ * @param version - The version to download
+ * @param osPlat - The operating system platform for which the Node.js binary was compiled.
+ * See https://nodejs.org/api/os.html#osplatform for more.
+ * @param osArch - The operating system CPU architecture for which the Node.js binary was compiled.
+ * See https://nodejs.org/api/os.html#osplatform for more.
+ * @returns The filename of the protocol buffer for the given release, platform and architecture.
+ *
+ */
+export function getFileName(
+  version: string,
+  osPlat: string,
+  osArch: string
+): string {
   // to compose the file name, strip the leading `v` char
   if (version.startsWith("v")) {
     version = version.slice(1, version.length);
@@ -126,13 +167,13 @@ function getFileName(version: string): string {
     return util.format("protoc-%s-win%s.zip", version, arch);
   }
 
-  const arch: string = osArch == "x64" ? "x86_64" : "x86_32";
+  const suffix = fileNameSuffix(osArch);
 
   if (osPlat == "darwin") {
-    return util.format("protoc-%s-osx-%s.zip", version, arch);
+    return util.format("protoc-%s-osx-%s.zip", version, suffix);
   }
 
-  return util.format("protoc-%s-linux-%s.zip", version, arch);
+  return util.format("protoc-%s-linux-%s.zip", version, suffix);
 }
 
 // Retrieve a list of versions scraping tags from the Github API

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -116,8 +116,8 @@ async function downloadRelease(version: string): Promise<string> {
 
 /**
  *
- * @param osArch - A string identifying the operating system platform for which the Node.js binary was compiled.
- * See https://nodejs.org/api/os.html#osplatform for possible values.
+ * @param osArch - A string identifying operating system CPU architecture for which the Node.js binary was compiled.
+ * See https://nodejs.org/api/os.html#osarch for possible values.
  * @returns Suffix for the protoc filename.
  */
 function fileNameSuffix(osArch: string): string {

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -192,11 +192,11 @@ async function fetchVersions(
 
   let tags: IProtocRelease[] = [];
   for (let pageNum = 1, morePages = true; morePages; pageNum++) {
-    let nextPage: IProtocRelease[] =
-      (await rest.get<IProtocRelease[]>(
-        "https://api.github.com/repos/protocolbuffers/protobuf/releases?page=" +
-          pageNum
-      )).result || [];
+    let p = await rest.get<IProtocRelease[]>(
+      "https://api.github.com/repos/protocolbuffers/protobuf/releases?page=" +
+        pageNum
+    );
+    let nextPage: IProtocRelease[] = p.result || [];
     if (nextPage.length > 0) {
       tags = tags.concat(nextPage);
     } else {

--- a/src/installer.ts
+++ b/src/installer.ts
@@ -147,7 +147,7 @@ function fileNameSuffix(osArch: string): string {
  * @param osPlat - The operating system platform for which the Node.js binary was compiled.
  * See https://nodejs.org/api/os.html#osplatform for more.
  * @param osArch - The operating system CPU architecture for which the Node.js binary was compiled.
- * See https://nodejs.org/api/os.html#osplatform for more.
+ * See https://nodejs.org/api/os.html#osarch for more.
  * @returns The filename of the protocol buffer for the given release, platform and architecture.
  *
  */

